features/fixtures/ios-swift-cocoapods/iOSTestApp/scenarios/Scenario.swift

```diff
@@ -0,0 +1,60 @@
+//
+// Created by Jamie Lynch on 06/03/2018.
+// Copyright (c) 2018 Bugsnag. All rights reserved.
+//
+
+import Foundation
+import Bugsnag
+
+internal class Scenario {
+
+    let config: BugsnagConfiguration
+
+    required init(config: BugsnagConfiguration) {
+        self.config = config
+    }
+
+
+    /**
+     * Sets a NOP implementation for the Session Tracking API, preventing delivery
+     */
+    func disableSessionDelivery() {
+        // TODO
+    }
+
+
+    /**
+     * Sets a NOP implementation for the Error Reporting API, preventing delivery
+     */
+    func disableReportDelivery() {
+        // TODO
+    }
+
+    /**
+     * Sets a NOP implementation for the Error Tracking API and the Session Tracking API,
+     * preventing delivery
+     */
+    func disableAllDelivery() {
+        self.disableReportDelivery()
+        self.disableSessionDelivery()
+    }
+
+    func generateException() -> NSException {
```

Comment:
Each individual scenario should specify its own exception rather than deriving it from a helper - this doesn't reflect how exception handling would work in a real app.